### PR TITLE
More docs

### LINKED ISSUE #23: Dokumentation
- [x] Problem umdefinieren/Use Case definieren: Wir sind Nutzer der App
- [x] Ziel der Challenge Deep Learning: Ja oder Nein, dokumentieren wieso Segment Based, Vorteile/Nachteile
- [x] Modelle einander gegenüberstellen
- [x] Lessons Learned schreiben: Mehr Personen wenn Person based Split

Models.md:
- [x] Currently hyperparams of the experiment we document the results of are completely unknown (kind of split etc). Should we link to the relevant experiments/pull requests? Also the resulting Confusion Matrices etc. are very hidden there.

### PR DESCRIPTION
**Pull request by gabrieltorresgamez**
References

https://github.com/Sensor-Based-Activity-Recognition/docs/issues/23

### CODE REVIEW

**Comment by gabrieltorresgamez** [comment]
Ich bi nöd all zu zfriede mit minere ergänzig abr bin grad bitz hilflos. Was sötti no id Zämmefassig hinzuefüege? sötti die Sache irgedwie/irgedwo anders dokumentiere?

File: Zusammenfassung.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Der Train-Val-Test-Split wurde entweder auf segmentebasierter Ebene oder durch einen personenbasierten Split durchgeführt, um die Leistung des Systems zu bewerten. Bei einem segmentbasierten Split wurden die Daten auf Segmentebene aufgeteilt, um eine detaillierte Bewertung der Aktivitätserkennung zu ermöglichen. Diese Methode ermöglicht die Präzisierung der Aktivitäten auf einzelne Segmente und die Erkennung spezifischer Aktivitätsmuster. Allerdings könnte sie den Einfluss individueller Unterschiede zwischen den Benutzern verringern. Ein personenbasierter Split hingegen teilt die Daten nach Personen auf und bewertet die Leistung des Systems anhand der Accuracy und des Micro F1-Scores. Diese Methode berücksichtigt individuelle Unterschiede zwischen den Benutzern und ermöglicht eine genauere Bewertung der Systemleistung für jede einzelne Person. Allerdings könnte sie die Präzision bei der Erkennung spezifischer Aktivitätsmuster verringern.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: > um eine detaillierte Bewertung der Aktivitätserkennung zu ermöglichen

Isch doch bi beidne ahsätz dr fall?

> Diese Methode berücksichtigt individuelle Unterschiede zwischen den Benutzern und ermöglicht eine genauere Bewertung der Systemleistung für jede einzelne Person.

Chumi nid drus.



Für mi isch das nid verständlich de abschnitt. Chasch de ganz absatz umschriebe:
Zersch erkläre, dass mir zwöi ahsätz fürs splitting hei. Denne seisch das de person-based für en app besser wär wo au benutzer installiere wo kei date contributed hei (smartphone/-watch hersteller) und me brucht vill date derzue. Das hei mir aber nid und üses ziel isch au nid en app für anderi persone z mache sondern en app für üs und drum hei mir en segment based gmacht.

File: Zusammenfassung.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# Methodik und Metriken
+
+Im Rahmen der Aktivitätserkennung wurden verschiedene Modelle verwendet, darunter das Convolutional Neural Network (CNN) und der Histogram Gradient Boosting Classifier (HGBC).
+
+Zur Bewertung der Modellleistung wurden die Accuracy und der Macro F1-Score verwendet. Die Accuracy gibt an, wie gut das Modell insgesamt die Aktivitäten klassifiziert, während der Macro F1-Score ein gewichtetes Mittelmass für die Präzision und den Recall aller Aktivitäten darstellt. Die Entscheidung, die Accuracy als Metrik zu verwenden, ermöglichte uns eine umfassende Bewertung und Vergleich der Gesamtleistung der Modelle. Eine hohe Accuracy deutet darauf hin, dass das Modell die Aktivitäten korrekt erkennt und klassifiziert. Der Macro F1-Score wurde ausgewählt, um sicherzustellen, dass die Leistung des Modells gleichmässig für alle Aktivitäten berücksichtigt wird. Er gewichtet die Präzision und den Recall für jede Aktivität gleich, unabhängig von der Datenverteilung. Diese Metrik ermöglichte uns eine Bewertung und Vergleich der Leistung der Modelle bei der Erkennung verschiedener Aktivitäten.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: > Zur Bewertung der Modellleistung wurden die Accuracy und der Macro F1-Score verwendet

Gemäss Konzept judge mir nach F1 und mir düe eifach au no d accuracy reporte.

File: Zusammenfassung.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+In diesem Projekt haben wir uns für einen segmentbasierten Split entschieden, da wir aufgrund der begrenzten Datenmenge von nur 10 Personen nicht genügend Daten hatten, um ein Modell zu entwickeln, das auf die breitere Masse abgestimmt ist. Daher lag unser Fokus auf der Entwicklung von Modellen, die auf unsere spezifische Leistung abgestimmt sind.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: Setz dr fokus vor begründig ufs argument, dass mir en app entwickle wo nur üsi activities classified. s andere darfsch scho erwähne, aber bitte ersch an 2ter stell.

File: Zusammenfassung.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# Modelle
+
+Ein ausführliche Übersicht aller Modelle kann man unter [Modelle](https://github.com/Sensor-Based-Activity-Recognition/docs/blob/main/Modelle.md) finden.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: Würde ich wie folgt anpassen:

Ein ausführliche Übersicht aller Modelle ~kann man~ **ist** unter [...] **zu** finden.

File: Zusammenfassung.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## CNN-Modell
+
+Das CNN-Modell wurde auf Grundlage des Papers von [Chen (2021)](https://github.com/Sensor-Based-Activity-Recognition/docs/blob/adaboost-docs/recherche/Chen_2021.md) entwickelt. Wie im Paper beschrieben, wurden Spectrogramme aus den Sensordaten mithilfe der Short-Time Fourier Transform erstellt. Die Implementierung des CNN erfolgte mit PyTorch und das Modell wurde geringfügig angepasst.
+
+Die Daten wurden aus der Datenbank extrahiert und auf 50Hz resampelt. Anschliessend wurden sie in 5-Sekunden-Fenster aufgeteilt. Während der STFT-Phase wurden die Spectrogram,e erstellt. Die Phase dvclive trainierte das Modell und die evaluate-Phase erstellte eine Konfusionsmatrix, um die Art der Fehler zu analysieren, die das CNN macht.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: > Spectrogram,e

sollte wohl _Spectrograme,_ heissen

File: Zusammenfassung.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+Die Architektur des Modells besteht aus vier Convolutional Layers, gefolgt von Pooling Layers. Nach dem Convolution-Teil folgt eine Global Average Pooling Layer und zwei Fully Connected Layers. Das Modell erzielte eine Test-Accuracy von über 90%.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: >, gefolgt von Pooling Layers

müsste heissen: 

, **jeweils** gefolgt von Pooling Layers

File: Zusammenfassung.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## HGBC-Modell
+
+Das HGBC-Modell verwendet den [HistGradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html) von sklearn. Als Eingabe für unsere Verarbeitungspipeline verwenden wir folgende Features: Accelerometer (X-, Y-, Z-Achse), Gyroskop (X-, Y-, Z-Achse), Magnetometer (X-, Y-, Z-Achse). Die Daten wurden resampelt, in 5-Sekunden-Segmente aufgeteilt und jedes Segment wurde in den Frequenzspektrumraum projiziert. Anschliessend wurde ein Train-Test-Split durchgeführt und das Modell wurde trainiert. Das HGBC-Modell erzielte eine beeindruckende Test-Accuracy von 98%.
```
Comment (florinbarbisch) [inline Zusammenfassung.md:35]: > Anschliessend wurde ein Train-Test-Split durchgeführt und das Modell wurde trainiert.

Würde ich beim CNN auch noch erwähnen. Sonst hat man das Gefühl, dass dies nur beim HGBC-Modell getan wird.

File: Zusammenfassung.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+## Vergleich der beiden Modelle
+
+Der Vergleich der beiden Modelle ergab, dass sowohl das CNN-Modell als auch das HGBC-Modell beeindruckende Ergebnisse erzielten. Das CNN-Modell zeigte seine Stärken bei der Erkennung von Aktivitäten aufgrund seiner Fähigkeit, spezifische Merkmale aus den Spectrograms zu extrahieren. Das HGBC-Modell hingegen erreichte eine hohe Genauigkeit, insbesondere bei der Erkennung von Aktivitäten mit einfachen Bewegungsmustern.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: Viiel zu kurz. Und nid würklich ussagechräftiger als ChatGPT.

Würd uf d fehler us dr konfusionsmatrix goh, und argumentiere, dass dank üsem guete preprocessing s HGBC besser isch. Do chasch au no allgmein druf ihgoh, wieso deep learning nid besser isch, respektive wies sich im ahsatz mit klassischem ML unterscheidet (eifach mal alli date ine schmeisse (deep learning) vs Features manuell extrahiere (ML)).

Vllt chasch do no chli reserach betriibe wenn klassisches ML superior zu deep learning isch und luege, was au ahwendig uf üses projekt het.

File: Zusammenfassung.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+## Vergleich der beiden Modelle
+
+Der Vergleich der beiden Modelle ergab, dass sowohl das CNN-Modell als auch das HGBC-Modell beeindruckende Ergebnisse erzielten. Das CNN-Modell zeigte seine Stärken bei der Erkennung von Aktivitäten aufgrund seiner Fähigkeit, spezifische Merkmale aus den Spectrograms zu extrahieren. Das HGBC-Modell hingegen erreichte eine hohe Genauigkeit, insbesondere bei der Erkennung von Aktivitäten mit einfachen Bewegungsmustern. 
+
+# Lessons Learned
+
+Eine wichtige Lesson Learned ist, dass man nicht ausschliesslich auf Deep Learning Modelle setzen sollte. Es ist auch wichtig, herkömmliche Machine Learning Modelle zu analysieren und in Betracht zu ziehen. Im Rahmen dieses Projekts konnten wir feststellen, dass einfache Modelle wie der HGBC in einigen Fällen sogar bessere Ergebnisse erzielen können als Deep Learning Modelle. Dies unterstreicht die Bedeutung der Auswahl des geeigneten Modells für das jeweilige Problem und die vorhandenen Daten.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: Hei mir do nid no meh sache gha?

Paar vorschläg vo mir
- [ ] nutze vo DVC (vllt no das mir mit stages baue übertriebe hei bevor mir es modell hei gha und jetzt vill unbruchti stages hei)
- [ ] segment- vs people-based split (split ist nicht immer einfach pro datenpunkt)
- [ ] why gradient boosting (said to be superior in many kaggle challenges)
- [ ] schwierigkeiten mit daten von unterschiedlichen sensore: **BENNNNN!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!**

File: Zusammenfassung.md
Comment (florinbarbisch) [inline Zusammenfassung.md:1]: Wenn wosch chasch no was übers team/Projectmanagement schriebe (flexe mit tools, dass alles guet abgloffe isch und mir eifach die beste si)

**Review by florinbarbisch — CHANGES_REQUESTED** [review]

File: Modelle.md [hunk 10]
```diff
@@ -66,6 +66,8 @@ The model achieves a Performance of over 90% on the test set on the Accuracy. He
 |  test_acc_epoch    |     0.922       |
 |  test_f1_epoch     |     0.915       |
 
+CML Report: https://github.com/Sensor-Based-Activity-Recognition/pipelines/pull/98#issuecomment-1564147778
```
Comment (florinbarbisch) [inline Modelle.md]: Markdown kennt [Links](https://github.com/Sensor-Based-Activity-Recognition/pipelines/pull/98#issuecomment-1564147778)!!!

File: Modelle.md [hunk 11]
```diff
@@ -148,3 +150,5 @@ The model achieves a Performance of over 98% on the test set on the Accuracy. He
 |:------------------:|----------------------------:|
 |  test_acc_epoch    |     0.98525       |
 |  test_f1_epoch     |     0.98363       |
+
+CML Report: https://github.com/Sensor-Based-Activity-Recognition/pipelines/pull/99#issuecomment-1564102269
```
Comment (florinbarbisch) [inline Modelle.md]: Markdown kennt [Links](https://github.com/Sensor-Based-Activity-Recognition/pipelines/pull/99#issuecomment-1564102269)!!!

File: Modelle.md (hunk 10 above)
Comment (gabrieltorresgamez) [reply Modelle.md]: ich bi aber rechts

**Comment by gabrieltorresgamez** [comment]
@florinbarbisch made some changes, whats your impression now?

File: Zusammenfassung.md [hunk 12]
```diff
@@ -8,21 +8,26 @@ Die Motivation für dieses Projekt ergibt sich aus der steigenden Popularität v
...
+Zur Bewertung der Modellleistung wurde der Macro F1-Score verwendet, jedoch haben wir diesen noch mit der Accuracy ergänzt. 
+- Wir haben Accuracy als Metrik genommen, um einen Gesamtüberblick über die Anzahl richtiger Klassifikationen zu erhalten. 
+- Wir haben den Macro F1-Score genommen, um festzustellen, ob ein Modell eine bestimmte Klasse bevorzugt. Wenn der F1-Score der Accuracy nahe liegt, deutet das darauf hin, dass unser Modell keine Klasse stark bevorzugt.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: > ob ein Modell eine bestimmte Klasse bevorzugt

lässt sich nicht ableiten. (Auch ist der F1 ein einziger Fakt).

Comment (florinbarbisch) [reply Zusammenfassung.md]: ok sehe was du sagen willst: "ob ein Modell alle Klasse gleich gut klassifizieren kann".

File: Zusammenfassung.md [hunk 13]
```diff
@@ -8,21 +8,26 @@ Die Motivation für dieses Projekt ergibt sich aus der steigenden Popularität v
...
+- Zusätzlich generieren wir aus Debugging-Gründen auch eine Decision Matrix.
```
Comment (florinbarbisch) [inline Zusammenfassung.md]: Würde sagen: falls F1 und accuracy nicht wie oben erwähnt nahe bei einander liegen: können wir die fehler in der Decision-Matrix untersuchen. 

Das Wort Debugging bezieht sich eher auf technische vorgänge und nicht aus menschlichen.

File: Zusammenfassung.md (hunk 6 above)
Comment (florinbarbisch) [reply Zusammenfassung.md]: Bitte anpassen @gabrieltorresgamez

**Review by florinbarbisch — CHANGES_REQUESTED** [review]

File: Zusammenfassung.md (hunk 13 above)
Comment (gabrieltorresgamez) [reply Zusammenfassung.md]: vorallem ist niemandem von uns aufgefallen, dass es eine confusion matrix und nicht eine decision matrix ist hahahah

**Comment by gabrieltorresgamez** [comment]
@buesst1 nice commit name

image

**Review by florinbarbisch — APPROVED** [review]
